core/inbox.go

```diff
@@ -39,7 +39,12 @@ type MessageProvider interface {
 
 // NewInbox constructs a new inbox.
 func NewInbox(pool *MessagePool, maxAgeRounds uint, chain InboxChainProvider, messages MessageProvider) *Inbox {
-	return &Inbox{pool: pool, maxAgeTipsets: maxAgeRounds, chain: chain}
+	return &Inbox{
+		pool:          pool,
+		maxAgeTipsets: maxAgeRounds,
+		chain:         chain,
+		messages:      messages,
```

Comment:
This should be `messageProvider` (or `msgGetter` or something like that). `messages` looks too much like a collection of messages.